mlflow/store/tracking/file_store.py

```diff
@@ -1319,3 +1326,372 @@ def _read_helper(root, file_name, attempts_remaining=2):
+            key: The string key of the tag.
+            value: The string value of the tag.
+        """
+        trace_info, trace_dir = self._get_trace_info_and_dir(request_id)
+        trace_info.tags.update({key: value})
+        self._write_dict_to_trace_sub_folder(
+            trace_dir, FileStore.TRACE_TAGS_FOLDER_NAME, {key: value}
+        )
+
+    def delete_trace_tag(self, request_id: str, key: str):
+        """
+        Delete a tag on the trace with the given request_id.
+
+        Args:
+            request_id: The ID of the trace.
+            key: The string key of the tag.
+        """
+        trace_info, trace_dir = self._get_trace_info_and_dir(request_id)
+        if key not in trace_info.tags:
+            raise MlflowException(
+                f"No tag with name: {key} in trace with request_id {request_id}.",
+                RESOURCE_DOES_NOT_EXIST,
+            )
+        trace_info.tags.pop(key)
+        _validate_tag_name(key)
+        tag_path = os.path.join(trace_dir, FileStore.TRACE_TAGS_FOLDER_NAME, key)
+        if not exists(tag_path):
+            raise MlflowException(
+                f"No tag with name: {key} in trace with request_id {request_id}.",
+                RESOURCE_DOES_NOT_EXIST,
+            )
+        os.remove(tag_path)
+
+    def delete_traces(
+        self,
+        experiment_id: str,
+        max_timestamp_millis: Optional[int] = None,
+        max_traces: Optional[int] = None,
+        request_ids: Optional[List[str]] = None,
+    ) -> int:
+        """
+        Delete traces based on the specified criteria.
+
+        - Either `max_timestamp_millis` or `request_ids` must be specified, but not both.
+        - `max_traces` can't be specified if `request_ids` is specified.
+
+        Args:
+            experiment_id: ID of the associated experiment.
+            max_timestamp_millis: The maximum timestamp in milliseconds since the UNIX epoch for
+                deleting traces. Traces older than or equal to this timestamp will be deleted.
+            max_traces: The maximum number of traces to delete.
+            request_ids: A set of request IDs to delete.
+
+        Returns:
+            The number of traces deleted.
+        """
+        # request_ids can't be an empty list of string
+        if max_timestamp_millis is None and not request_ids:
+            raise MlflowException(
+                "Either `max_timestamp_millis` or `request_ids` must be specified.",
+                INVALID_PARAMETER_VALUE,
+            )
+        if max_timestamp_millis and request_ids:
+            raise MlflowException(
+                "Only one of `max_timestamp_millis` and `request_ids` can be specified.",
+                INVALID_PARAMETER_VALUE,
+            )
+        if request_ids and max_traces is not None:
+            raise MlflowException(
+                "`max_traces` can't be specified if `request_ids` is specified.",
+                INVALID_PARAMETER_VALUE,
+            )
```

Comment:
Do we need these restrictions for file store? These just seem to be enforced by Databricks backend design (the shared design with Run table), but probably we don't need to enforce it for this storage. I indeed removed this in SQLAlchemyStore, because they are not intuitive and may confuse users. WDYT?